### PR TITLE
[2870] Add an in-app changelog

### LINKED ISSUE #2870: Add an in-app changelog
- Add an in-app changelog that's triggered when opening the app after an upgrade (or series of upgrades) to let the user know what's new from the last time he/she used the app
- Proposed library to use: https://github.com/TonnyL/WhatsNew
- We can either hardcode it, or we can have a JSON file on the server, or somewhere in the code like @VaiTon suggested

### PR DESCRIPTION
**Pull request by naivekook**
####  Description
Add new screen to show all our new functionality to users in a fancy way.

#### Related issues
Linked issue [ #{2870}](https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/2870)

#### Screenshots
photo5253948161242804625

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/feature/changelog/ChangelogDialog.kt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                                version.items.forEach { item ->
+                                    itemList.add(ChangelogListItem.Item("- $item"))
+                                }
+                            }
+                            itemList
+                        }
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(
+                                { items -> recyclerView.adapter = ChangelogAdapter(items) },
+                                { throwable -> throwable.printStackTrace() }
```
Comment (naivekook) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/feature/changelog/ChangelogDialog.kt]: Do we need to log this somehow?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/MainActivity.kt [hunk 2]
```diff
@@ -420,6 +421,10 @@ class MainActivity : BaseActivity(), NavigationDrawerListener {
         binding.bottomNavigationInclude.bottomNavigation.selectNavigationItem(0)
         binding.bottomNavigationInclude.bottomNavigation.installBottomNavigation(this)
         handleIntent(intent)
+
+        if (isFlavors(AppFlavors.OFF)) {
```
Comment (naivekook) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/MainActivity.kt:425]: only for OFF now

**Comment by naivekook** [comment]
@teolemon @VaiTon I need help to add "changelog" files to crowdin

**Review by VaiTon — APPROVED** [review]
Code-wise it's ok. @teolemon have you tested?

**Comment by teolemon** [comment]
Not yet, but it looks good. My biggest concern was architecture wise (so that we don't show the message twice, and that we have a simple translation path)

**Comment by VaiTon** [comment]
@naivekook isn't there anything to use the resources created by fastlane with the `versioncode.txt` file?

**Comment by naivekook** [comment]
@VaiTon sorry I don't understand the question
artifacts from fastlane are not included in apk

**Comment by teolemon** [comment]
Ok, let's merge and debug in the internal release 👍  YOLO :-)

**Pull request merged**